activesupport/lib/active_support/cache/streaming_compressor.rb

```diff
@@ -0,0 +1,110 @@
+# frozen_string_literal: true
+
+require "zlib"
+require "concurrent/executor/cached_thread_pool"
+require "active_support/core_ext/numeric/bytes"
+
+module ActiveSupport
+  module Cache
+    # Serializes object and compresses serialized representation in streaming way.
+    # This avoids storing whole copy of serialized intermediate representation,
+    # like this happens in `Zlib.deflate(Marshal.dump(value))`.
+    module StreamingCompressor
+      DEFLATE_BUFFER_SIZE = 16.kilobytes.to_i
+      INFLATE_CHUNK_SIZE = 4.kilobyte.to_i
+
+      extend self
+
+      # Dumps value into string. Returns nil if `:compress_threshold` option
+      # is set and serialized value is less than given value.
+      def dump(value, **options)
```

Comment:
Seems like the only use for `**options` is for `compress_threshold` , `**` operator is expensive.
Although it will not look as clean, explicitly passing `compress_threshold` will be quicker and given that it's only one argument now, seems reasonable.
https://gist.github.com/ohaddahan/f59aa5a261cb56e7b1ca96fb97a8e3fb
@schneems recently improved `try` by avoid unneeded `* splat` operator.

Reply:
I'll fix it. Thanks!